tests/io/test_base.py

```diff
@@ -155,6 +155,18 @@ def test_generate_header(self, scheduler):
             )
             scheduler.generate_header(res)
 
+        with pytest.raises(
+            ValueError,
+            match=r"The following keys are not present in the template: option32, processes-per-node. "
+            r"Check the template in .*MyScheduler.header_template.*option3 instead of option32. processes_per_node instead of processes-per-node",
+        ):
```

Comment:
Should it also assert that "instead of option32" is not in the error message ?

Reply:
Maybe I am misunderstanding or there is a type? Because "instead of option32" is in the error message.

Reply:
Ah yes indeed :D I thought it was not. So the idea was to have a key that is not present in the template and that has no match either. Don't know if it's really needed anyway.

Reply:
I added this check to the previous test with "titi"